src/main.rs

```diff
@@ -128,12 +132,19 @@ impl Resources {
         const HIT_FX: &str = include_str!("../assets/fxses/hit.json");
         const EXPLOSION_FX: &str = include_str!("../assets/fxses/explosion.json");
         const LIFE_UI_FX: &str = include_str!("../assets/fxses/life_ui_explosion.json");
-
+        const CANNONBALL_HIT_FX: &str = include_str!("../assets/fxses/canonball_hit.json");
+        const EXPLOSION_PARTICLES: &str = include_str!("../assets/fxses/explosion_particles.json");
+        const SMOKE_FX: &str = include_str!("../assets/fxses/smoke.json");
         let hit_fxses = EmittersCache::new(nanoserde::DeJson::deserialize_json(HIT_FX).unwrap());
         let explosion_fxses =
             EmittersCache::new(nanoserde::DeJson::deserialize_json(EXPLOSION_FX).unwrap());
         let life_ui_explosion_fxses =
             EmittersCache::new(nanoserde::DeJson::deserialize_json(LIFE_UI_FX).unwrap());
+        let fx_cannonball_hit =
```

Comment:
Will cannonball fx be used in too many weapons to be loaded here, in global resources list? 

(it may be, but from the first look it belongs to weapon's fx list more)